app/models/lookup_key.rb

```diff
@@ -158,6 +158,11 @@ def path_elements
     end
   end
 
+  def overriden_on_host?(host)
+    return false if host.is_a?(Hostgroup)
```

Comment:
hostgroup also uses `puppetclasses/_class_parameters` but you would need to change the match you are looking for.